pytest_mojo/plugin.py

```diff
@@ -66,7 +71,9 @@ def collect(self):
             if line.startswith(TEST_ITEM_PREFIX):
                 if cur_item is not None:
                     yield MojoTestItem.from_parent(
-                        self, name=cur_item, spec=dict(stdout=item_stdout, code=0)
+                        self,
+                        name=cur_item,
+                        spec=dict(stdout=item_stdout, code=process.returncode),
```

Comment:
Here I think we need to keep return `code=0`. I know this conditional logic is not very clear, and I struggled with it.  The situation here is that the line starts with the test item prefix AND the cur_item is not None,  therefore what is happening is the **second** test item is starting. So we know that the first item must succeeded. Definitely open to ways to improve this or make it clearer- maybe just more code comments.

Reply:
I know what you mean, I was conflicted by this one too. Looks to me like it would be a shoot in our foot later. As we are not properly solving the underlying problem, I am moving it back to 0, and I am going to do a refactor in a follow-up PR.